pkg/tfbridge/schema.go

```diff
@@ -1022,33 +1022,61 @@ func CleanTerraformSchema(tfs map[string]*schema.Schema) map[string]*schema.Sche
 }
 
 // CoerceTerraformString coerces a string value to a Go value whose type is the type requested by the Terraform schema
-// type. Returns an error if the string can't be successfully coerced to the requested type.
-func CoerceTerraformString(schType schema.ValueType, stringValue string) (interface{}, error) {
+// type or the Pulumi SchemaInfo. We prefer the SchemaInfo overrides as it's an explicit call to action over the
+// Terraform Schema. Returns an error if the string can't be successfully coerced to the requested type.
+func CoerceTerraformString(schType schema.ValueType, schemaOverride *SchemaInfo, stringValue string) (interface{}, error) {
+	// check for the override and use that over terraform if available
+	// we do this to ensure that we are following the explicit call to action of the override
+	if schemaOverride != nil && schemaOverride.Type != "" {
+		switch schemaOverride.Type {
+		case "boolean":
```

Comment:
Are the names of these types defined anywhere?  Like is there any existing sense in which `"boolean"` is meaningful as a value of this (vs. for example `"bool"`)?  Should we define the legal values of these types somewhere?

Reply:
This is interesting actually. I only used primitive types for these as we define these as strings in our Pulumi SchemaInfo - if you think changing these to be an actual type then I'd be behind that

Reply:
Would it make sense to use the same types as in schema? AFAIK, those are `boolean`, `integer`, and `number`.